pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -0,0 +1,467 @@
+package sti
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+	dockerpkg "github.com/openshift/source-to-image/pkg/docker"
+	"github.com/openshift/source-to-image/pkg/errors"
+	"github.com/openshift/source-to-image/pkg/tar"
+	"github.com/openshift/source-to-image/pkg/util"
+)
+
+type postExecutorStepContext struct {
+	// id of the previous image that we're holding because after committing the image, we'll lose it.
+	// Used only when build is incremental and RemovePreviousImage setting is enabled.
+	// See also: rememberPreviousImageStep and removePreviousImageStep
+	previousImageID string
+
+	// Container id that will be committed.
+	// See also: commitBuilderImageStep/commitRuntimeImageStep
+	containerID string
+
+	// See also: commitBuilderImageStep/commitRuntimeImageStep
+	location string
+
+	// Container id after committing.
+	// See also: commitBuilderImageStep/commitRuntimeImageStep and reportAboutSuccessStep
+	imageID string
+
+	// Labels that will be passed to a callback.
+	// It's labels that is used during commit of builder image (or runtime image if it's used).
+	// See also: commitBuilderImageStep/commitRuntimeImageStep and invokeCallbackStep
+	labels map[string]string
+}
+
+type postExecutorStep interface {
+	execute(*postExecutorStepContext) error
+}
+
+type rememberPreviousImageStep struct {
+	builder *STI
+	docker  dockerpkg.Docker
+}
+
+func (step *rememberPreviousImageStep) execute(ctx *postExecutorStepContext) error {
+	if step.builder.incremental && step.builder.config.RemovePreviousImage {
+		glog.V(1).Info("Executing step: remember previous image")
+		ctx.previousImageID = step.getPreviousImage()
+	} else {
+		glog.V(3).Info("Skipping step: remember previous image")
+	}
+	return nil
+}
+
+func (step *rememberPreviousImageStep) getPreviousImage() string {
+	previousImageID, err := step.docker.GetImageID(step.builder.config.Tag)
+	if err != nil {
+		glog.V(0).Infof("error: Error retrieving previous image's (%v) metadata: %v", step.builder.config.Tag, err)
+		return ""
+	}
+	return previousImageID
+}
+
+type removePreviousImageStep struct {
+	builder *STI
+	docker  dockerpkg.Docker
+}
+
+func (step *removePreviousImageStep) execute(ctx *postExecutorStepContext) error {
+	if step.builder.incremental && step.builder.config.RemovePreviousImage {
+		glog.V(1).Info("Executing step: remove previous image")
```

Comment:
@bparees After reading @smarterclayton logging rules, I'm starting to doubt whether we should show name of executed post step on level 1.